### PR TITLE
Move LogREST into pod/rest package

### PR DESCRIPTION
**Pull request by feihujiang**
Code refactoring,  make the Pod REST implementation less coupled to etcd and moving them out of pkg/registry/pod/etcd into pkg/registry/pod/rest.

### CODE REVIEW

**Comment by brendandburns** [comment]
LGTM, but needs rebase.

**Comment by feihujiang** [comment]
@brendandburns I rebased, and move a test from pkg/registry/pod/etcd into pkg/registry/pod/rest. Please review.

File: pkg/registry/pod/rest/log.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	"k8s.io/kubernetes/pkg/api/validation"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	etcdgeneric "k8s.io/kubernetes/pkg/registry/generic/etcd"
+	genericrest "k8s.io/kubernetes/pkg/registry/generic/rest"
+	"k8s.io/kubernetes/pkg/registry/pod"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+// LogREST implements the log endpoint for a Pod
+// TODO: move me into pod/rest - I'm generic to store type via ResourceGetter
```
Comment (brendandburns) [inline pkg/registry/pod/rest/log.go]: nit: delete TODO, it's done!

**Comment by brendandburns** [comment]
LGTM, one nit.

**Pull request merged**